File: QytetetJava/src/modeloqytetet/Casilla.java

package modeloqytetet;

 public abstract class Casilla {
    protected int numeroCasilla;
    protected TipoCasilla tipo;

    public Casilla (int numeroCasilla, TipoCasilla tipo) {
        this.numeroCasilla = numeroCasilla;
        this.tipo = tipo;
    }

    public TipoCasilla getTipo() {
        return tipo;
    }
   

    public int getNumeroCasilla() {
        return numeroCasilla;
    }
    
    boolean soyEdificable() {
        return tipo == TipoCasilla.CALLE;
    }  
    
    abstract void setTitulo(TituloPropiedad titulo);
    abstract void setNumHoteles(int numHoteles);
    abstract void setNumCasas(int numCasas);
    abstract public TituloPropiedad getTitulo();
    abstract TituloPropiedad asignarPropietario(Jugador jugador);
    abstract int calcularValorHipoteca();
    abstract int cancelarHipoteca();
    abstract int cobrarAlquiler();
    abstract int edificarCasa();
    abstract int edificarHotel();
    abstract boolean estaHipotecada();
    abstract int getCoste();
    abstract int getCosteHipoteca();
    abstract int getNumCasas();
    abstract int getNumHoteles();
    abstract int getPrecioEdificar();
    abstract int hipotecar();
    abstract int precioTotalComprar();
    abstract boolean propietarioEncarcelado();
    abstract boolean sePuedeEdificarCasa(int factorEspeculador);
    abstract boolean sePuedeEdificarHotel(int factorEspeculador);
    abstract boolean tengoPropietario();
    abstract int venderTitulo();
    abstract void asignarTituloPropiedad();
    
}

File: QytetetRuby/lib/casilla.rb
#encoding: utf-8
require_relative "tablero"
require_relative "jugador"
require_relative "titulo_propiedad"
require_relative "tipo_casilla"

module ModeloQytetet
  class Casilla
    attr_accessor :numeroCasilla, :tipo
    
    def initialize(numeroCasilla, tipo)
      @numeroCasilla = numeroCasilla
      @tipo = tipo     
    end
    
    def get_tipo
      @tipo
    end
    
    def get_numero_casilla
      @numeroCasilla
    end
    
    def soy_edificable
      @tipo == TipoCasilla::CALLE
    end
    
    

    def to_s
      "Casilla\n { numeroCasilla= #{@numeroCasilla} , tipo= #{@tipo} }\n"
    end
    
    
  end
end
